### ISSUE TITLE
Option to not show DAG import error in the UI

### ISSUE THREAD

**Issue by sudarshan2906** [apache/airflow#17044]
**Description**

It would be good if there is a option in cfg to hide the DAG import error from the UI. Or only to show it to Admin or the role which has permission to view it.

![image](https://user-images.githubusercontent.com/25225354/125887496-b92431df-f8c2-463f-8254-5f0962162dd7.png)

**Use case / motivation**

We have a multi team setup in airflow. And its very annoying if a team is able to see other DAG import teams error. 
The simple fix would be to just disable it with a config property

**Are you willing to submit a PR?**

Not sure

**Comment by eladkal** [comment]
Do you have [DAG-level permission ](https://airflow.apache.org/docs/apache-airflow/stable/security/access-control.html#dag-level-permissions)set? If set and we are showing errors to users without permission on that DAG then it's probably a bug rather than a feature request.

**Comment by mik-laj** [comment]
> permission on that DAG

If the file cannot be loaded, the DAG object cannot be created. We only have access to the DAG File, so no access control covers this yet. We don't have ACL for DAG File yet, so it is not bug.

**Comment by sudarshan2906** [comment]
@mik-laj Yes agreed, Its not a bug and it is not possible with the current setup. What do think about disabling it via a config parameter?
Just to not display it in the UI but we can fetch it with ImportError API or we can see it in the logs.

**Comment by bbovenzi** [comment]
The API will only send import errors if the user has `AccessView.IMPORT_ERRORS`, so one should be able to change who has that permission and therefore choose when import errors show up
